src/XPurchase/VirtoCommerce.ExperienceApiModule.XOrder/Authorization/CanAccessOrderAuthorizationHandler.cs

```diff
@@ -0,0 +1,40 @@
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using VirtoCommerce.OrdersModule.Core.Model;
+using VirtoCommerce.Platform.Security.Authorization;
+
+namespace VirtoCommerce.ExperienceApiModule.XOrder.Authorization
+{
+    public sealed class CanAccessOrderAuthorizationRequirement : PermissionAuthorizationRequirement
+    {
+        public CanAccessOrderAuthorizationRequirement() : base("CanAccessOrder")
+        {
+        }
+    }
+
+    public class CanAccessOrderAuthorizationHandler : PermissionAuthorizationHandlerBase<CanAccessOrderAuthorizationRequirement>
```

Comment:
I think better will be extending this requirement https://github.com/VirtoCommerce/vc-module-order/blob/dev/src/VirtoCommerce.OrdersModule.Web/Authorization/OrderAuthorizationHandler.cs with a new logic to allow view and edit any order that belongs to you (has the same CustomerId as your UserId) and just reuse this authorization for all xOrder requests.

Reply:
Decided to not extend existing OrderModuleHandler in favor of implement own handler in the XAPI. But it should be extended with search criteria/request support.